SofaImGui/src/SofaImGui/windows/Components.h

```diff
@@ -0,0 +1,37 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU General Public License as published by the Free  *
+* Software Foundation; either version 2 of the License, or (at your option)   *
+* any later version.                                                          *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for    *
+* more details.                                                               *
+*                                                                             *
+* You should have received a copy of the GNU General Public License along     *
+* with this program. If not, see <http://www.gnu.org/licenses/>.              *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/simulation/Node.h>
+
+namespace sofaimgui
+{
+    class ImGuiGUIEngine;
+
+    class Components
+    {
+    public:
+        static void showComponents(const char* const& windowNameComponents, bool& isComponentsWindowOpen);
+
+    };
```

Comment:
I don't recommend a design based on a class for such a simple thing. You can implement a free function in the `Components` namespace. For example:

```cpp
namespace Components
{
    void showComponents(const char* const& windowNameComponents, bool& isComponentsWindowOpen);
}
```

However, in SOFA we try to keep a match between the namespaces and the folder structure. In this case, I would suggest

```cpp
namespace windows
{
    void showComponents(const char* const& windowNameComponents, bool& isComponentsWindowOpen);
}
```

which means that you will have to call it with `windows::showComponents(....)`.

Also add a brief doxygen comment to describe the function.

Reply:
@lamriaimen my comment applies for all your similar class in the windows folder

Reply:
Both please